os/tools/zipdownload.sh

```diff
@@ -0,0 +1,140 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+
+# Usages
+
+USAGE="
+
+USAGE: ${0} [-d] [FILE-NAME]
+Zip binary, debug files like elf and map and flash tool files
+
+  -d         include debug files, elf and map into zip file
+  FILE-NAME  zip file name
+               when not configured, "TizenRT_\<board-name\>_\<commit_hash\>.tar" used
+"
+
+# Parsing arguments
+
+unset DEBUG
+unset FILENAME
+
+while [ ! -z "$1" ]; do
+	case "$1" in
+	--help )
+		echo "$USAGE"
+		exit 0
+		;;
+	-d )
+		DEBUG=y
+		;;
+	*)
+		if [ ! -z "${FILENAME}" ]; then
+			echo ""
+			echo "File name defined twice"
+			echo "$USAGE"
+			exit 1
+		fi
+		FILENAME=$1
+		;;
+	esac
+	shift
+done
+
+# Zip command & option
+
+TAR=tar
+TAROPT=-cf
+
+# Get folder path
+
+OSTOOLDIR=`test -d ${0%/*} && cd ${0%/*}; pwd`
+TOPDIR="${OSTOOLDIR}/../.."
+OSDIR="${TOPDIR}/os"
+BUILDDIR="${TOPDIR}/build"
+BINDIR="${BUILDDIR}/output/bin"
+
+# Get fixed file
+
+CONFIG=${OSDIR}/.config
+if [ ! -f ${CONFIG} ]; then
+	echo "No .config file"
+	exit 1
+fi
+MAKEFILE="${OSDIR}/Makefile ${OSDIR}/Makefile.unix ${OSDIR}/Makefile.win ${OSDIR}/Directories.mk ${OSDIR}/FlatLibs.mk ${OSDIR}/LibTargets.mk ${OSDIR}/Make.defs ${OSDIR}/tools/Config.mk"
+SCRIPTFILE="${OSDIR}/tools/download.sh ${OSDIR}/tools/download.bat"
+
+BINARY=${BINDIR}/tinyara.bin
+if [ ! -f "${BINARY}" ]; then
+	echo "No binary file"
+	exit 1
+fi
+
+unset ELF
+unset MAP
+
+# Get configuration
+
+source ${CONFIG}
+
+BOARD=${CONFIG_ARCH_BOARD}
+
+# Get file according to .config
+
+SCRIPTDIR="${BUILDDIR}/configs/${BOARD}"
+TOOLCHAINDEFS="${OSDIR}/arch/${CONFIG_ARCH}/src/${CONFIG_ARCH_FAMILY}/Toolchain.defs"
+
+# Get GIT information (if not provided on the command line)
+
+if [ -z "${FILENAME}" ]; then
+	GITINFO=`git log 2>/dev/null | head -1`
+	if [ -z "${GITINFO}" ]; then
+		echo "GIT version information is not available"
+		BUILD=NA
+	else
+		BUILD=`echo ${GITINFO} | cut -d' ' -f2`
+		BUILD=`echo ${BUILD:0:7}`
+		if [ -z "${BUILD}" ]; then
+			echo "GIT build information not found"
+			BUILD=NA
+		fi
+	fi
+
+	FILENAME=tizenrt_${BOARD}_${BUILD}.tar
+fi
+
+if [ "${DEBUG}" == "y" ]; then
+	ELF=${BINDIR}/tinyara
+	MAP=${BINDIR}/tinyara.map
+fi
+
+# Board specific
+
+unset FLASHTOOLDIR
+
+if [ "${BOARD}" == "artik05x" ]; then
+	FLASHTOOLDIR="${BUILDDIR}/tools/openocd"
+	BOARDSPECIFIC="${BUILDDIR}/configs/artik053 ${BUILDDIR}/configs/artik053s ${BUILDDIR}/configs/artik055s ${BINDIR}/tinyara_head.bin"
```

Comment:
Also, Don't we need to copy files required for make_romfs()?

Reply:
1. I will add romfs.img but I don't want to copy whole build/output/bin. Because debug files, elf and map are too big. That is not necessary to all. That's why I make an option, -d.
2. I will modify that logic to make a romfs image. Currently, that occures during executing download script. Actually making romfs image is not a step of downloading so that I will move it into Makefile.

Reply:
OK. 
artik05x_download.sh has support for flashing **_OTA(ota.bin)_**. Don't we need to add it?

Reply:
@manoharahk I added ota also.